src/main.c

```diff
@@ -97,6 +113,13 @@ int main(int argc, char* argv[]) {
         case 't':
             opts.threads = strtoul(optarg, NULL, 10);
             break;
+        case 'm':
+            opts.method = atoi(optarg);
+            if (opts.method < 0 || opts.method > 2) {
+                fprintf(stderr, "Método de reserva inválido. Usando el método por defecto.\n");
+                opts.method = 0;
+            }
```

Comment:
`atoi` no es adecuado para comprobación de errores, por ejemplo, pasar un string que no sea un entero hace que `atoi` devuelva 0, cómo sabemos entonces si el usuario eligió 0 o ingresó basura? Si quieres realizar comprobación de errores en serio, utiliza `strtol`. Y por otro lado, si vamos a fallar, fallamos, no usamos un valor por defecto.
```suggestion
            int* endptr = NULL;
            opts.method = strtol(optarg, &endptr, 10);
            if (*endptr != \0) {
                fprintf(stderr, "Método de reserva inválido: '%s'.\n", optarg);
                return -1;
            }

            if (opts.method < 0 || opts.method > 2) {
                fprintf(stderr, "Método de reserva inválido.\n");
                return -1;
            }
```
https://linux.die.net/man/3/strtol